signer/tests/integration/withdrawal_reject.rs

```diff
@@ -0,0 +1,453 @@
+        Error::WithdrawalRejectValidation(ref err) => {
+            assert_eq!(err.error, WithdrawalRejectErrorMsg::RequestMissing)
+        }
+        err => panic!("unexpected error during validation {err}"),
+    }
+
+    testing::storage::drop_db(db).await;
+}
+
+/// For this test we check that the `RejectWithdrawalV1::validate` function
+/// returns a withdrawal validation error with a RequestNotFinal message
+/// when bitmap in the transaction does not match what our records would
+/// create for the bitmap.
+#[tokio::test]
+async fn reject_withdrawal_validation_not_final() {
+    // Normal: this generates the blockchain as well as a transaction
+    // sweeping out the funds for a withdrawal request. This is just setup
+    // and should be essentially the same between tests.
+    let db = testing::storage::new_test_database().await;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+    let (rpc, faucet) = regtest::initialize_blockchain();
+    let setup = TestSweepSetup::new_setup(&rpc, &faucet, 1_000_000, &mut rng);
+
+    // Normal: the signer follows the bitcoin blockchain and event observer
+    // should be getting new block events from bitcoin-core. We haven't
+    // hooked up our block observer, so we need to manually update the
+    // database with new bitcoin block headers.
+    backfill_bitcoin_blocks(&db, rpc, &setup.sweep_block_hash).await;
+
+    // Normal: we take the sweep transaction as is from the test setup and
+    // store it in the database.
+    setup.store_sweep_tx(&db).await;
+
+    // Normal: we need to store a row in the dkg_shares table so that we
+    // have a record of the scriptPubKey that the signers control.
+    setup.store_dkg_shares(&db).await;
+
+    // Normal: the request and how the signers voted needs to be added to
+    // the database. Here the bitmap in the withdrawal request object
+    // corresponds to how the signers voted.
+    setup.store_withdrawal_request(&db).await;
+    setup.store_withdrawal_decisions(&db).await;
+
+    // Generate the transaction and corresponding request context.
+    let (reject_withdrawal_tx, req_ctx) = make_withdrawal_reject(&setup);
+
+    // This should not return an Err.
+    let ctx = TestContext::builder()
+        .with_storage(db.clone())
+        .with_first_bitcoin_core_client()
+        .with_mocked_stacks_client()
+        .with_mocked_emily_client()
+        .build();
+
+    // Generate more blocks then backfill the DB
+    let mut hashes = faucet.generate_blocks(5);
+    let last = hashes.pop().unwrap();
+    backfill_bitcoin_blocks(&db, rpc, &last).await;
+
+    let validation_result = reject_withdrawal_tx.validate(&ctx, &req_ctx).await;
+    match validation_result.unwrap_err() {
+        Error::WithdrawalRejectValidation(ref err) => {
+            assert_eq!(err.error, WithdrawalRejectErrorMsg::RequestNotFinal)
+        }
+        err => panic!("unexpected error during validation {err}"),
+    }
+
```

Comment:
After this we should generate one more block and check that things validate then.

Reply:
This test is just happy_path with one less block, but seems reasonable to mine the extra block and make really sure that it now passes.

Reply:
And of course it's not passing once I add the block and expect validation :P  Not sure if it's test fixture or the validation is wrong.

Reply:
Oh lol the problem is that I reused the `req_ctx`, which means that the `chain_tip` didn't change even though I mined an extra block ⛓️ 

I knew there was a reason to get the chain tip inside `validate`.  Though I guess if we can't trust `req_ctx` we have bigger problems.

Fixed now.